3.5/troubleshooting-reduce-memory-footprint.md

```diff
@@ -0,0 +1,353 @@
+- User-defined AQL functions
+
+Note that JavaScript / V8 can be disabled for DB-Server and Agency nodes in a
+cluster without these limitations. They apply to single server instances. They
+also apply to Coordinator nodes, but you should not disable V8 on Coordinators
+because certain cluster operations depend on it.
+
+CPU usage
+---------
+
+We can’t really reduce CPU usage, but the number of threads running in parallel.
+Again, you should not do this unless there are very good reasons, like an
+embedded system. Note that this will limit the performance for concurrent
+requests, which may be okay for a local development system with you as only 
+user.
+
+The number of background threads can be limited in the following way:
+
+``` 
+--arangosearch.threads-limit 1
+--rocksdb.max-background-jobs 4
+--server.maintenance-threads 2 
+--server.maximal-threads 4
+--server.minimal-threads 1
+```
+
+In general, the number of threads is selected to fit the machine. However, each
+thread requires at least 8 MB of stack memory. By sacrificing some performance
+for parallel execution it is possible to reduce this.
+
+This option will make logging synchronous:
+
+``` 
+--log.force-direct true
+```
+
+This is not recommended unless you only log errors and warnings.
+
+Examples
+--------
+
+In general, you should adjust the read buffers and edge cache on a standard
+server. If you have a graph use-case, you should go for a larger edge cache. For
+ example, split the memory 50:50 between read buffers and edge cache. If you
+have no edges then go for a minimal edge cache and use most of the memory for
+the read buffers.
+
+For example, if you have a machine with 40 GByte of memory and you want to
+restrict ArangoDB to 20 GB of that, use 10 GB for the edge cache and 10 GB for
+the read buffers if you use graph features.
+
+Please keep in mind that during query execution additional memory will be used
+for query results temporarily. If you are tight on memory, you may want to go
+for 7 GB each instead.
+
+If you have an embedded system or your development laptop, you can use all of
+the above settings to lower the memory footprint further. For normal operation,
+especially production, these settings are not recommended.
+
+Linux System Configuration
+--------------------------
+
+The main deployment target for ArangoDB is Linux. As you’ve learned above
+ArangoDB and its innards work a lot with memory. Thus its vital to know how
+ArangoDB and the Linux kernel interact on that matter. The linux kernel offers
+several modes of how it will manage memory. You can influence this via the proc
+filesystem, the file `/etc/sysctl.conf` or a file in `/etc/sysctl.conf.d/` which
+your system will apply to the kernel settings at boot time. The settings as
+named below are intended for the sysctl infrastructure, meaning that they map
+to the `proc` filesystem as `/proc/sys/vm/overcommit_memory`.
+
+After studying the kernel documentation our former recommendation was to set
```

Comment:
this is not good imho - this is correct for a blog post, but not for the guide - my opinion

Reply:
shortened the sentence by a big deal